src/transformations.jl

```diff
@@ -8,16 +8,30 @@ abstract type AbstractDataTransform end
 
 Apply transformation `t` to vector or matrix `x` in place.
 """
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,1}) = transform!(x, t, x)
-transform!(t::AbstractDataTransform, x::AbstractArray{<:Real,2}) = transform!(x, t, x)
+function transform!(t::AbstractDataTransform, x::AbstractVecOrMat{<:Real}; dims::Union{Integer,Nothing}=nothing)
+    if dims == 1
+        return transform!(x, t, x)
+    elseif dims == 2
+        return transform!(x', t, x')'
```

Comment:
Return `x` instead of calling `'`, which would return an `Adjoint` object wrapping it instead.

Reply:
This still applies:
```suggestion
        transform!(x', t, x')
        return x
```
Same below. Tests should be adapted of course.